packages/flame/lib/src/components/mixins/has_game_ref.dart

```diff
@@ -17,9 +17,10 @@ mixin HasGameRef<T extends Game> {
   set gameRef(T gameRef) {
     _gameRef = gameRef;
     if (this is BaseComponent) {
+      // TODO(luan) this is wrong, should be done using propagateToChildren
```

Comment:
on a HasGameRef > no game ref > HasGameRef chain this will not work.

I will revisit this when trying to generify propageToChildren later.